test/Dockerfile

```
@@ -1,37 +1,45 @@
 # syntax=docker.io/docker/dockerfile-upstream:1.12.1-labs
 
+#### prepare tools
+FROM rust:1.85 AS tools
+# foundry
+RUN set -eux; \
+  apt-get update && apt-get install -y --no-install-recommends curl ca-certificates git; \
+  curl -L https://foundry.paradigm.xyz | bash; \
+  ~/.foundry/bin/foundryup --install stable; \
+  install -Dm755 ~/.foundry/bin/* /usr/local/bin/
```

Applying the following diff got the Docker build to work for me.

```suggestion
ARG DEBIAN_FRONTEND=noninteractive
RUN <<EOF
set -eu
apt-get update && apt-get install -y --no-install-recommends curl ca-certificates git
curl -L https://foundry.paradigm.xyz | bash
~/.foundry/bin/foundryup --install stable
install -Dm755 ~/.foundry/bin/* /usr/local/bin/
EOF
```